aws/rust-runtime/aws-config/src/credential_process.rs

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Credentials Provider for external process
+
+use crate::json_credentials::{parse_json_credentials, JsonCredentials};
+use aws_types::credentials::{future, CredentialsError, ProvideCredentials};
+use aws_types::{credentials, Credentials};
+use std::process::Command;
+
+/// Credentials Provider
+#[derive(Debug)]
+pub struct CredentialProcessProvider {
+    credential_process: String,
+}
+
+impl ProvideCredentials for CredentialProcessProvider {
+    fn provide_credentials<'a>(&'a self) -> future::ProvideCredentials<'a>
+    where
+        Self: 'a,
+    {
+        future::ProvideCredentials::new(self.credentials())
+    }
+}
+
+impl CredentialProcessProvider {
+    /// Create new [`CredentialProcessProvider`]
+    pub fn new(credential_process: String) -> Self {
+        Self { credential_process }
+    }
```

Comment:
I think it'd be good to refactor this to take a `Vec<String>` rather than parsing in here (`moving the parsing out into aws/rust-runtime/aws-config/src/profile/credentials/repr.rs`) but curious to hear thoughts.

Alternatively, the `new` here could return a `Result` and do the parsing up front.

Reply:
I'd support having new return a result here

Reply:
Ended up refactoring this to take the Java SDK's approach of just passing the whole string through `sh -c` / `cmd.exe /C` in [14ad7a9](https://github.com/awslabs/smithy-rs/pull/1356/commits/14ad7a9084a5047d8863f6038e675663738cadd4) to avoid needing to do any parsing.